### ISSUE TITLE
Add aria-describedby or aria-label support to `CustomSelectControl`

### ISSUE THREAD

**Issue by getdave** [WordPress/gutenberg#22988]
**Is your feature request related to a problem? Please describe.**
Often options in a select dropdown make more sense in context of a wider setting. Therefore sometimes it is useful to be able to add additional context to each option in the dropdown.

For an example of this please see https://github.com/WordPress/gutenberg/pull/18869#discussion_r435117857. There is a dropdown indicating that the user can either

1. Create a menu from one of the options listed.
2. Create a menu from existing Pages.
3. Create a blank menu.

Therefore there are 3 types of action in the dropdown but these are only disambiguate via visual devices. It would be preferable to add additional context to the "menu" options via `aria-describedby` which could say something like 

> Create a new menu using items from "Primary Menu"

...or something to that effect. 

Whilst this isn't possible with native `<select>` dropdowns, it is possible with the `CustomSelectControl` options.

**Describe the solution you'd like**
Let's consider adding support for [the `aria-describedby` attribute](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-describedby_attribute) on the `CustomSelectControl` dropdown options in order that we can provide additional context for each item. 

**Describe alternatives you've considered**
We could simply require folks to use fully descriptive labels for options but sometimes this isn't sufficient.

**Followups**

Once merged we should follow up by updating the Navigation block placeholder to support this new attribute:

https://github.com/WordPress/gutenberg/pull/18869#discussion_r435117857

**Comment by talldan** [comment]
Here's a screenshot of the UI in question:
<img width="597" alt="Screenshot 2020-06-10 at 6 59 08 pm" src="https://user-images.githubusercontent.com/677833/84260355-7ced1d80-ab4c-11ea-8f78-f4bb76f4bb78.png">

My feeling is that 'Main Menu' is a little ambiguous. It refers to the name of a menu that the user will use to create the menu. When moving through the options using a screenreader it might be particularly ambiguous, so an `aria-label` could be used to enhance what's read, e.g. "Create using 'Main Menu'".

@getdave On the PR comment you mentioned:
> I've looked into this further and aria-label should only be used when there is no visible label text available. IN this case we have visible label text which is the option's text itself.

> An alternative is aria-describedby which provides a reference to a location which can provide additional context. If we enhance CustomSelectControl to allow this attr we can have an element which reacts to the selection providing additional context to users of assistive technologies.

I think `aria-description` tends to be used for secondary information in addition to the existing label, whereas here I think the visual label needs to be overridden, which is why I suggested using `aria-label`. Open to feedback on that, so I've added `Needs Accessibility Feedback`.

**Comment by tellthemachines** [comment]
Adding screen reader specific text to the individual options in the select doesn't feel right at all. If the options read as ambiguous, we should fix their text for all users. Screen reader users should be clearly informed of the purpose of the dropdown by means of its label, and from there they should have enough context to navigate the options. What I propose:

* Let's improve the text inside the `<label>`. Currently, the `<label>` is visually hidden and reads "select to create from pages, existing menu or empty". That's not awfully clear.
* Let's make the label visible, and remove the  visible `<div>` that reads "Create a navigation from all existing pages, or choose a menu.": we should only need one set of clear, visible and screen reader accessible instructions.
* Let's remove the "placeholder" option altogether and set a sensible default as the first value.

Suggestions very welcome for copy improvement 🙂

**Comment by tellthemachines** [comment]
Closing this issue in favour of #22999
